src/lbaf/IO/lbsMeshBasedVisualizer.py

```diff
@@ -36,28 +37,27 @@ def __init__(
         # Assign logger to instance variable
         self.__logger = logger
 
-        # Make sure that quantity of interest name was passed
-        if not isinstance(qoi, list) or not len(qoi) or not (
-            qoi_name := qoi[0]) or not isinstance(qoi_name, str):
+        # Make sure that quantity of interest names were passed
+        if not isinstance(qoi_request, list) or not len(qoi_request) > 2 or not (
+            rank_qoi := qoi_request[0]) or not isinstance(
+            rank_qoi, str) or not (object_qoi := qoi_request[2]) or not isinstance(
+            object_qoi, str):
```

Comment:
This is quite hard to read. Can you extract it into a separate helper (returning `bool`) or split it up into named boolean variables?